### ISSUE TITLE
SDDM recently broke loging in i3

### ISSUE THREAD

**Issue by Zimmi48** [NixOS/nixpkgs#21353]
## Issue description

When logging in i3 from the SDDM login manager, I get an xterm window instead, with the latest nixos-16.09 channel.

### Steps to reproduce

I was able to reproduce the problem on two different machines:

With the following configuration.nix:

```
{ config, pkgs, ... }:

{
  # Basic hardware and boot-loader config

  # Enable the X11 windowing system.
  services.xserver = {
    enable = true;

    displayManager.sddm.enable = true;
    windowManager.i3.enable = true;
  };

  # Basic user declaration
  };
}
```

With version 16.03.1299.a8e0739 (Emu), everything works fine.
With version 16.09.1324.1dd0fb6 (Flounder), I cannot log in i3 anymore.

To note: on the other machine where I had the same problem I did not save the version number before the upgrade but I'm pretty sure that it was > 16.09 so I think the problem appeared in a recent update of this channel.

Switching to another login manager (LightDM) fixes the problem.

This may be related to #15722 but I'm not too sure about that, plus this issue seem pretty old.

**Comment by jokogr** [comment]
@Zimmi48 could you report back the versions of sddm and i3 you are using from both channels?

**Comment by Zimmi48** [comment]
@jokogr I've just git bisected the problem and I've seen it was introduced by commit ca1e677 (upgrading SDDM from 0.13.0 to 0.14.0).

**Comment by jokogr** [comment]
@Zimmi48 I have managed to reproduce your issue and it appears that no sessions work with SDDM at all.

As a workaround, you could create a `~/.xsession` file where you put `exec i3`.

In my system, it appears that `/nix/store/cajai5ysa23jdxc10nma7s22r1k9sn6p-xsession /nix/store/cajai5ysa23jdxc10nma7s22r1k9sn6p-xsession 'none + i3'` is trying to be executed, which is wrong. It is strange to me that other display managers don't have an issue with that. But anyway, even if I set `SessionCommand` in `/etc/sddm.conf` to `/run/current-system/sw/bin/sh`, it is not working.

If anyone has a suggestion, please hop in.

**Comment by vcunat** [comment]
Why was that release bump actually picked into the stable release?

**Comment by bachp** [comment]
Does this issue still exist with 17.03?

**Comment by Zimmi48** [comment]
@bachp Yes, I just tested. Note that I'm not using SDDM anymore and I suppose no one uses SDDM + i3 but still, it would be good to fix it.

**Comment by gnidorah** [comment]
@Zimmi48 
> I suppose no one uses SDDM + i3 
I would like to switch to sddm to try this https://unix.stackexchange.com/questions/209274/how-do-i-tell-systemd-logind-that-the-session-is-not-idle-without-using-a-deskt

**Comment by gnidorah** [comment]
This fixed i3+ssdm for me 
https://github.com/NixOS/nixpkgs/pull/23264
```bash
git fetch https://github.com/nixos/nixpkgs pull/23264/head
git cherry-pick 3cca092a35e883c041c9a1f0092042452eeb22f0
```

**Comment by Mic92** [comment]
fixed in master and cherry-picked to 17.03 in 40a5498c3f9558b1f033e17ebd65c21c56e4f0ba
